lib/xmlrpc/server.rb

```diff
@@ -532,6 +532,49 @@ def http_write(body, status, header)
 end
 
 
+# Implements a XML-RPC server, which works with Rack
+class RackServer < BasicServer
+
+  # This method processes a XML-RPC method call and sends the answer
+  # back to the client.
+  def call(env)
+    length = env['CONTENT_LENGTH'].to_i
+
+    return http_error(405, "Method Not Allowed") unless env['REQUEST_METHOD'] == "POST"
+    return http_error(400, "Bad Request")        unless parse_content_type(env['CONTENT_TYPE']).first == "text/xml"
+    return http_error(411, "Length Required")    unless length > 0
+
+    req = Rack::Request.new(env)
```

Comment:
Do we need to require `rack` in `initialize`?

Reply:
I would guess this is already available when you run this as Rack middleware, so that should only be required if you run this rack app without rack. This seems like an unlikely scenario to me.

Reply:
BTW, this is a Rack application not Rack middleware, right?